server/src/main/java/org/jboss/as/server/services/net/LocalDestinationOutboundSocketBindingAddHandler.java

```diff
@@ -71,6 +79,30 @@ protected void performRuntime(final OperationContext context, final ModelNode op
 
         final PathAddress address = PathAddress.pathAddress(operation.get(OP_ADDR));
         final String outboundSocketName = address.getLastElement().getValue();
+
+        context.addStep(new OperationStepHandler() {
+            @Override
+            public void execute(OperationContext context, ModelNode operation) throws OperationFailedException {
+                final String currentSocketBindingGroupName = address.getParent().getLastElement().getValue();
+                final Set<ResourceEntry> children = context.readResourceFromRoot(PathAddress.EMPTY_ADDRESS)
+                        .getChildren(ModelDescriptionConstants.SOCKET_BINDING_GROUP);
+                for (Resource.ResourceEntry re : children) {
+                    if (re.getName().equalsIgnoreCase(currentSocketBindingGroupName)) {
+                        for (Resource.ResourceEntry entry : re.getChildren(ModelDescriptionConstants.SOCKET_BINDING)) {
```

Comment:
You need to check REMOTE_DESTINATION_OUTBOUND_SOCKET_BINDING as well.

Instead of iterating over the children, all re.getChildrenNames(SOCKET_BINDING) and then do a simple contains check.